### ISSUE TITLE
config file and input data for hrnet classification task

### ISSUE THREAD

**Issue by chengzg** [microsoft/O-CNN#108]
Hi,

@wang-ps  Firstly thanks for sharing the code for your great work.

I would like to test the mid-net with the classification task as described in the paper, would you mind sharing the config file and also the pre-processed data as well? Thanks.

Thanks & Regards,
Spencer

**Comment by chengzg** [comment]
Hi,

I also want to train the mid-net with my own dataset, how should i prepare the dataset particularly on the label value? Inside convert_tfrecords.py, i assign each asset with a unique value(incremental one). However, I get the following error when i start the training with the generated tf record data: 

octree_property_op.cc:92: check_failed: channel_ == channel (1 vs. 0) The specified channel_ is wrong. Property name: label

Would appreciated if any help is provided. Thanks.

Thanks & Regards,
Spencer

**Comment by wang-ps** [comment]
Thanks for the interest in our project. Please be patient, I will update the code soon. 
Here are some tips to do the experiments:
1. For the classification task, the [network](https://github.com/microsoft/O-CNN/blob/master/tensorflow/script/network_hrnet.py)  is ready to run.  You can write the yaml config file for HRNet, referring to the yaml file [here](https://github.com/microsoft/O-CNN/blob/master/tensorflow/script/configs/cls_octree_test.yaml). Note that the default upright direction of ModelNet40 is Z-axis, and the default upright direction of ShapeNet is Y-axis. When using the pretrained weight, you should align the models of ModelNet to Y-axis.

2. To prepare the data, you can first try to run with our data. And compare your data with ours. The error you provided is caused by missing of point labels. Please refer to [here](https://github.com/microsoft/O-CNN/blob/master/docs/data_preparation.md#custom-data)  to provide the pre-point labels when writing the `points` files.

**Comment by chengzg** [comment]
Hi,

@wang-ps, Thanks very much for your reply, really appreciated.
According to the comments at the file [here](https://github.com/microsoft/O-CNN/blob/master/docs/data_preparation.md#custom-data), the point labels could be empty.
![image](https://user-images.githubusercontent.com/5573089/98619647-a2ba6080-233e-11eb-8157-15adea8d5372.png)

From your answer, sounds like I have to provide the point labels? So what kind of point label should I provide? Is it the patch segment number that the point belongs to or some other point labels? Thanks very much.

Thanks & Regards,
Spencer

**Comment by wang-ps** [comment]
@chengzg  I have released the code for classification just now. Please pull the latest code and have a try.

> I have to provide the point labels?
The point label for instance discrimination is the correponding patch index.
